dali/pipeline/operators/reader/loader/coco_loader.cc

```diff
@@ -356,6 +346,27 @@ void ParseAnnotationFilesHelper(std::vector<std::string> &annotations_filename,
           image_id_pairs.push_back(std::make_pair(image_file_name, id));
           image_id_to_wh.insert(std::make_pair(id, std::make_pair(width, height)));
         }
+      } else if (0 == strcmp(key, "categories")) {
+        RAPIDJSON_ASSERT(r.PeekType() == kArrayType);
+        r.EnterArray();
+        int id = -1;
+        while (r.NextArrayValue()) {
+          if (r.PeekType() != kObjectType) {
+            continue;
+          }
+          r.EnterObject();
+          while (const char* internal_key = r.NextObjectKey()) {
+            if (0 == strcmp(internal_key, "id")) {
+              id = r.GetInt();
+            } else {
+              r.SkipValue();
+            }
+          }
+          if (id != -1) {
```

Comment:
Should we fail or there could be a category without an ID in the valid COCO?

Reply:
:+1: